src/aiida/tools/dumping/facades/profile.py

```diff
@@ -0,0 +1,74 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""ProfileDumper facade."""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, cast
+
+from aiida import load_profile
+from aiida.common.log import AIIDA_LOGGER
+from aiida.manage.configuration.profile import Profile
+from aiida.tools.dumping.config import DumpConfig
+from aiida.tools.dumping.engine import DumpEngine
+from aiida.tools.dumping.utils.paths import (
+    DumpPaths,
+    get_default_profile_dump_path,
+)
+
+if TYPE_CHECKING:
+    pass
+
+
+logger = AIIDA_LOGGER.getChild('tools.dumping.facades.profile')
+
+
+class ProfileDumper:
+    """Dumps all data in an AiiDA profile."""
+
+    def __init__(self, profile: Profile | str | None = None, config=None, output_path=None) -> None:
+        """_summary_
+
+        :param profile: _description_, defaults to None
+        :param config: _description_, defaults to None
+        :param output_path: _description_, defaults to None
+        """
+        self.profile: Profile = ProfileDumper._load_profile(profile)
+        self.config: DumpConfig | None = config
+
+        if output_path is None:
+            default_path = get_default_profile_dump_path(self.profile)
+            self.dump_paths = DumpPaths(parent=Path.cwd(), child=default_path)
+        else:
+            self.dump_paths = DumpPaths.from_path(output_path)
+
+        self.engine = DumpEngine(config=self.config, dump_paths=self.dump_paths)
+
+    @staticmethod
+    def _load_profile(profile: str | Profile | None) -> Profile:
+        """Load the AiiDA profile from string or Profile object."""
+
+        # TODO: See if this is necessary, and works in the first place for a different profile
+        if isinstance(profile, str):
+            loaded_profile = load_profile(profile=profile, allow_switch=True)
+        elif isinstance(profile, Profile):
+            loaded_profile = profile
+        else:
+            from aiida.manage import get_manager
+
+            manager = get_manager()
+            loaded_profile = cast(Profile, manager.get_profile())
+
+        assert loaded_profile is not None
```

Comment:
Error with custom message is better, but at this point you cannot really give descriptive message. Why should be loaded_profile None? Following the types there is no way it can be None. Since this is a public class, I can see that the `profile` input is not following the types, then I would change the branching above and add
```python
elif profile is None:
    ...
else:
    raise TypeError(f"Provided type of profile is {type(profile)} but only Profile | str | None are supported.")